collie/metrics.py

```diff
@@ -196,37 +236,50 @@ def mapk(targets: csr_matrix,
 
 
 def mrr(targets: csr_matrix,
-        user_ids: Union[np.array, torch.tensor],
+        user_or_item_ids: Union[np.array, torch.tensor],
         preds: Union[np.array, torch.tensor],
-        k: Optional[Any] = None) -> float:
+        k: Optional[Any] = None,
+        negative_sample_type: Literal['item', 'user'] = 'item') -> float:
     """
     Calculate the mean reciprocal rank (MRR) of the input predictions.
 
     Parameters
     ----------
     targets: scipy.sparse.csr_matrix
         Interaction matrix containing user and item IDs
-    user_ids: np.array or torch.tensor
-        Users corresponding to the recommendations in the top k predictions
+    user_or_item_ids: np.array or torch.tensor
+        Users or items corresponding to the recommendations in the top k predictions
     preds: torch.tensor
-        Tensor of shape (n_users x n_items) with each user's scores for each item
+        Tensor of shape (n_users x n_items) with each user's scores for each item or
+        tensor of shape (n_items x n_users) with each item's scores for each user
     k: Any
         Ignored, included only for compatibility with ``mapk``
+    negative_sample_type: str
+        Type of negative sampling the Interaction matrix, ``targets``, used
 
     Returns
     -------
     mrr_score: float
 
     """
-    predicted_items = preds.topk(preds.shape[1], dim=1).indices
-    labeled = _get_labels(targets, user_ids, predicted_items, device=preds.device)
-
-    # weighting each 0/1 by position so that topk returns index of *first* postive result
-    position_weight = 1.0/(
-        torch.arange(1, targets.shape[1] + 1, device=preds.device)
-        .repeat(len(user_ids), 1)
-        .float()
-    )
+    device = preds.device
+    predictions = preds.topk(preds.shape[1], dim=1).indices
+    labeled = _get_labels(targets, user_or_item_ids, predictions, device, negative_sample_type)
```

Comment:
[Optional] I realize the original code wasn't written this way, but my new coding philosophy is to be explicit with keyword arguments as much as possible. I think it's a lot clearer (and less error-prone) to write function calls like this: 

```python
foo(a=a, b=b, c=c, d=d)
```

instead of 

```python
foo(a, b, c, d)
```

If you have energy and want to, I think it would be nice to do that for all of our internal function calls in this file. If not, feel free to ignore.

Reply:
I think I covered all the cases of this, but please call out any I missed in the next review - happy to edit!